lib/rules/label-content-name-mismatch-matches.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Applicability:
+ * Rule applies to any element that has
+ * a) a semantic role that is `widget` that supports name from content
```

Comment:
You're not checking if the semantic role is widget. When you do, please note that the "type" property in the lookup table is not the same as it is in the ARIA spec. We'll need to figure something out about that. Maybe blacklist the ones that have type: widget that aren't widgets according to ARIA? I'm not sure if those things can be changed without breaking a bunch of stuff unfortunately.